howso/attribute_maps.amlg

```diff
@@ -195,6 +195,136 @@
+	;   which their deviations are stored under. For each group of shared deviations, they are stored under the first sorted feature in their group.
+	;   e.g., a group of features with shared devations is [A, B, C], creates an mapping of A -> A, B -> A, C -> A
+	; shared_deviations_features_lists <- list of lists of shared deviation feature groups
+	#!ProcessSharedDeviationsMap
+	(let
+		(assoc
+			shared_deviations_features_lists
+				(map
+					(lambda
+						;sort the list so duplicates can be removed
+						(sort (append [(current_value 1)] (get shared_deviations_map [(current_value 1) "shared_deviations"]) ))
+					)
+					(indices shared_deviations_map)
+				)
+			shared_deviations_key_map (assoc)
+		)
+
+		;get rid of duplicate groups
+		(assign (assoc shared_deviations_features_lists (values shared_deviations_features_lists (true))))
+
+		(map
+			(lambda (let
+				;each group of shared deviations features is already sorted
+				(assoc current_shared_features_list (current_value 1))
+				(map
+					(lambda
+						(accum
+							(assoc shared_deviations_key_map
+								(associate (current_value 2) (first current_shared_features_list))
+							)
+						)
+					)
+					current_shared_features_list
+				)
+			))
+			shared_deviations_features_lists
+		)
+
+		(assoc
+			"shared_deviations_key_map" shared_deviations_key_map
+			"shared_deviations_features_lists" shared_deviations_features_lists
+		)
+	)
+
+	;Validates the necessary parameters for shared deviations, making sure that each feature only belongs to one group of shared deviations,
+	; and that no shared deviations features are unique nominals.
+	#!ValidateSharedDeviations
+	(let
+		(assoc
+			invalid_repeated_shared_deviations_features (list)
+			invalid_unique_shared_deviations_features (list)
+			error_message (list)
+			shared_deviations_features_flattened (list)
+		)
+
+		;check to make sure that each feature only shows up once in any shared deviations groups
+		;flatten all of the values in the groups of shared deviations
+		(map
+			(lambda (accum (assoc
+				shared_deviations_features_flattened (current_value 1)
+			)))
+			(get shared_deviations_map "shared_deviations_features_lists")
+		)
+
+		;sort the flattened list
+		(assign (assoc shared_deviations_features_flattened (sort shared_deviations_features_flattened)))
+		;see if any value appears twice in a row, indicating a duplicate value
+		(map
+			(lambda
+				(let
+					(assoc current_shared_deviation_feature (current_value 1))
+					(if
+						(=
+							current_shared_deviation_feature
+							(get shared_deviations_features_flattened (+ (current_index) 1))
+						)
+						(accum (assoc invalid_repeated_shared_deviations_features current_shared_deviation_feature ))
+					)
+				)
+			)
+			shared_deviations_features_flattened
+		)
+
+		;check to see if any features in a shared deviations group is a unique nominal
+		(map
```

Comment:
instead do a filter over the indices of the "shared_deviations_key_map" where the (lambda) is the your condition you are using within the (if). 

also instead of (contains_value (indices ...)), you could use (contains_index)